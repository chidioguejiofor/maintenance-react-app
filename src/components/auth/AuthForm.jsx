import { Form, Container, Header, Icon, Message } from 'semantic-ui-react';
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import TopNav from '../Header';

const AuthForm = ({
  formType, handleSubmit, success, messageTitle,
  loading, handleChange, inputs, errorMessages,
  links
}) => (
  <div id="auth-form">
    <TopNav />
    <Container>
      <div className="form">
        <Form
          loading={loading}
          onSubmit={handleSubmit}
          error={!!errorMessages.length}
          success={success}
        >

          <Header as="h2" icon textAlign="center">
            <Icon name="user circle" />
            <Header.Content>{formType}</Header.Content>
          </Header>
          <Message
            header={messageTitle}
            error={!!errorMessages.length}
            success={success}
            list={errorMessages}
            hidden={!success && !errorMessages.length}
          />
          {inputs.map((input, index) => (
            <Form.Input
              key={index}
              label={input.placeholder}
              placeholder={input.placeholder}
              name={input.name}
              value={input.value}
              onChange={handleChange}
              type={input.type}
            />
          ))}
          <Form.Button
            disabled={success}
            positive
            content="Submit"
          />
          {links.map((linkObj, index) => (
            <span key={index} className="link">
              <Link to={linkObj.to}>{linkObj.caption}</Link>
            </span>

          ))}
        </Form>
      </div>

    </Container>
  </div>


);

AuthForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  inputs: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)).isRequired,
  formType: PropTypes.string,
  errorMessages: PropTypes.arrayOf(PropTypes.string),
  success: PropTypes.bool,
  messageTitle: PropTypes.string,
  loading: PropTypes.bool,
  links: PropTypes.arrayOf(PropTypes.object),
};
AuthForm.defaultProps = {
  formType: 'Sign Up',
  errorMessages: [],
  success: true,
  messageTitle: '',
  loading: false,
  links: [],
};
export default AuthForm;
