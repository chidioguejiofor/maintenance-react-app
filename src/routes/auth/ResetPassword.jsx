import React from 'react';


const ResetPassword = () => (
  <h1>This is the reset password</h1>
);

export default ResetPassword;
