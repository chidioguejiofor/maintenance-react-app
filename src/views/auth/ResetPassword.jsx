import React, { Component } from 'react';
import Form from '../../components/Form';
import Header from '../../components/Header';
/**
 *contains logic for the LoginPage.
 */
class SignUp extends Component {
  /**
   * this creates a new instance of Login class.
   * The super constructor is called here and the state is initialized
   * @param {object} props contains the props sent from its parent
   */
  constructor(props) {
    super(props);
    this.state = {
      password: '',
      email: '',
      success: false,
    };

    // bind methods
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = () => {};
  }


  /**
   * this handles all the change events that occurs in this Login
   * component
   * @param {obejct} e contains info about the event
   * @param {object} param1 contains the value and other properties
   * of the targe
   * @returns {void} handles the event and returns nothing
   */
  handleChange(e, { name, value }) {
    this.setState({ [name]: value });
  }


  /**
   * this renders a presentation component with values gotten from state
   *@returns {JSX} returns JSX to be rendered in the DOM
   */
  render() {
    const { password, success, email } = this.state;
    const inputs = [
      {
        name: 'email',
        placeholder: 'Email',
        value: email,
        type: 'email'
      },
      {
        name: 'password',
        placeholder: 'New Password',
        value: password,
        type: 'password'
      },
      {
        name: 'confirm-password',
        placeholder: 'Confirm Password',
        value: password,
        type: 'password'
      },

    ];

    const errorMessages = [];
    const links = [
      { caption: 'Sign Up', to: '/signup' },
      { caption: 'Login', to: '/login' }
    ];
    return (
      <div>
        <Header />
        <Form
          formTitle="Reset Password"
          errorMessages={errorMessages}
          success={success}
          inputs={inputs}
          handleChange={this.handleChange}
          handleSubmit={this.handleSubmit}
          links={links}
        />
      </div>

    );
  }
}
export default SignUp;
