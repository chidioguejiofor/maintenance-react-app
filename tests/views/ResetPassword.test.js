import React from 'react';
import { shallow } from 'enzyme';
import { Form, Input } from 'semantic-ui-react';
import chai from 'chai';
import ResetPassword from '../../src/views/auth/ResetPassword';
import AuthForm from '../../src/components/auth/AuthForm';

const getMounted = (options = {}) => shallow(<ResetPassword {...options} />);
describe('Testing AuthForm Component', () => {
  describe('the components when inputs property are empty rendered in the DOM', () => {
    it('should match snapshot', () => {
      const mountedObj = getMounted();
      expect(mountedObj).toMatchSnapshot();
    });

    it('should render an AuthForm', () => {
      const mountedObj = getMounted();
      chai.expect(mountedObj.find(AuthForm).length)
        .to.equal(1);
      chai.expect(
        mountedObj.find(AuthForm)
          .prop('inputs')
          .length
      )
        .to.equal(3);
    });
  });

  describe('the methods in the component', () => {
    describe('handleChange', () => {
      it('should update state when called', () => {
        const stateSpy = jest.spyOn(ResetPassword.prototype, 'setState');
        const spy = jest.spyOn(ResetPassword.prototype, 'handleChange');
        const mountedObj = getMounted();

        mountedObj
          .instance()
          .handleChange(null, { name: 'username', value: 'Test Username' });
        expect(spy)
          .toHaveBeenCalledTimes(1);
        expect(stateSpy)
          .toHaveBeenCalledTimes(1);
      });
    });
  });
});
