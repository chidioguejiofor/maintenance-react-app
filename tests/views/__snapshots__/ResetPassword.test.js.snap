// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AuthForm Component the components when inputs property are empty rendered in the DOM should match snapshot 1`] = `
<div>
  <Header />
  <AuthForm
    errorMessages={Array []}
    formTitle="Reset Password"
    handleChange={[Function]}
    handleSubmit={[Function]}
    iconUrl=""
    inputs={
      Array [
        Object {
          "name": "email",
          "placeholder": "Email",
          "type": "email",
          "value": "",
        },
        Object {
          "name": "password",
          "placeholder": "New Password",
          "type": "password",
          "value": "",
        },
        Object {
          "name": "confirm-password",
          "placeholder": "Confirm Password",
          "type": "password",
          "value": "",
        },
      ]
    }
    links={
      Array [
        Object {
          "caption": "Sign Up",
          "to": "/signup",
        },
        Object {
          "caption": "Login",
          "to": "/login",
        },
      ]
    }
    loading={false}
    messageTitle=""
    submitLoading={false}
    success={false}
    type="auth"
  />
</div>
`;
